pkg/controller/nodeipam/ipam/cidr_allocator.go

```diff
@@ -89,7 +89,7 @@ type CIDRAllocator interface {
 }
 
 // New creates a new CIDR range allocator.
-func New(kubeClient clientset.Interface, cloud cloudprovider.Interface, nodeInformer informers.NodeInformer, allocatorType CIDRAllocatorType, clusterCIDRs []*net.IPNet, serviceCIDR *net.IPNet, secondaryServiceCIDR *net.IPNet, nodeCIDRMaskSize int) (CIDRAllocator, error) {
+func New(kubeClient clientset.Interface, cloud cloudprovider.Interface, nodeInformer informers.NodeInformer, allocatorType CIDRAllocatorType, clusterCIDRs []*net.IPNet, serviceCIDR *net.IPNet, secondaryServiceCIDR *net.IPNet, nodeCIDRMaskSize []int) (CIDRAllocator, error) {
```

Comment:
Can we change New to take a struct? the parameter list is getting quite long:

```
type CIDRAllocatorParams struct {
  ClusterCIDRs []*net.IPNet
  ServiceCIDR *net.IPNet
  SecondaryServiceCIDR *net.IPNet
  NodeCIDRMaskSize []int
}

func New(kubeClient ..., cloud ..., nodeInformer ...,   allocatorType..., p CIDRAllocatorParams) ...
```

Reply:
done!